internal/status/immediatestatus.go

```diff
@@ -60,7 +60,22 @@ func (o *StatusObserver) OnNotify(status types.StatusEventArgs) error {
 	o.goalStateEventMap.Store(status.StatusKey, status.TopLevelStatus)
 	return o.OnDemandNotify()
 }
-
+func IsEqualStatusItem(statusItem1 types.StatusItem, statusItem2 types.StatusItem) bool {
+	if statusItem1.Version != statusItem2.Version ||
+		statusItem1.TimestampUTC != statusItem2.TimestampUTC {
+		return false
+	}
+	if statusItem1.Status.Name != statusItem2.Status.Name ||
+		statusItem1.Status.Operation != statusItem2.Status.Operation ||
+		statusItem1.Status.Status != statusItem2.Status.Status ||
```

Comment:
For these 3 checks, can the case differ (lowerCase vs upper ) ? same string, but different cases would be deemed not equal